golang/internal/config/config.go

```diff
@@ -4,11 +4,18 @@ import (
 	"time"
 )
 
-// configuration defaults
-// convention is we have ALL the options here, but where defaults should be
+// Configuration defaults
```

Comment:
I would add a simple, inline, textual flow diagram.
```
//
// application startup -> load Configuration (read environment variables) -> unwrap variables (custom type)
//
// the last step is a public SetValue function defined on the custom type
// similar to json/yaml deserializing
```

Or similar, wherever you see fit.